src/main/java/org/ow2/proactive/catalog/repository/BucketRepository.java

```diff
@@ -46,6 +46,8 @@
 public interface BucketRepository extends JpaRepository<BucketEntity, Long>, JpaSpecificationExecutor<BucketEntity>,
         QueryDslPredicateExecutor<BucketEntity> {
 
+    BucketEntity findFirstByBucketName(String bucketName);
```

Comment:
Since bucket name must be unique, it is better to name the method like findOneByBucketName instead of findFirst...,   there cannot be several buckets having same name